calico.py

```diff
@@ -140,6 +140,17 @@ def __init__(self, network_config, env):
         Path in which to search for CNI plugins.
         """
 
+        self.running_under_mesos = bool(self.network_config.get("args", {}). \
```

Comment:
Any risk of both `running_under_mesos` and `running_under_k8s` being set at the same time?  I think it can only happen in a really unlikely user error case, so should be fine, but ideally I'd put some sort of checking mechanism in place to make sure we're not hitting weird code paths we didn't intend to ever be hit.